website/docs/language/data-sources/index.mdx

```diff
@@ -78,18 +78,26 @@ in more detail in the following sections.
 
 ## Data Resource Behavior
 
-If the query constraint arguments for a data resource refer only to constant
-values or values that are already known, the data resource will be read and its
-state updated during Terraform's "refresh" phase, which runs prior to creating a plan.
-This ensures that the retrieved data is available for use during planning and
-so Terraform's plan will show the actual values obtained.
-
-Query constraint arguments may refer to values that cannot be determined until
-after configuration is applied, such as the id of a managed resource that has
-not been created yet. In this case, reading from the data source is deferred
-until the apply phase, and any references to the results of the data resource
-elsewhere in configuration will themselves be unknown until after the
-configuration has been applied.
+Terraform aims to read data resources during the planning phase whenever
+possible, but in order to achieve a correct order of operations Terraform
+defers reading a data resource until the apply phase in the following
+situations:
```

Comment:
I'm trying to make this flow better and include the tidbit about announcing it in the plan up here, as I think it's relevant. Ignore this is you don't like this edit....

```suggestion
Terraform reads data resources during the planning phase when possible and announces in the plan when it must defer reading resources until the apply phase to preserve the order of operations. Terraform defers reading data resources in the following situations:
```

Reply:
I took all of your suggestions here exactly as you wrote them, except that I kept the "but" instead of "and" in the opening sentence here because to me it more clearly read that we're about to describe exceptions to the rule. Please let me know if that doesn't feel right to you!

Reply:
I'm going to optimistically go ahead and merge this now just because I'm blocking the start of a release process, but I'll be happy to revise this in a subsequent PR if you don't like it! Let me know. :grinning:

Reply:
ahh sorry that I didn't catch it fast enough. I trust you on this :-) Thank you for letting me know, and I'm glad we were able to get this out!